students/PeterB/lesson02/Assignment/charges_calc.py

```diff
@@ -0,0 +1,144 @@
+
+    stream_handler = logging.StreamHandler()
+    stream_handler.setFormatter(formatter)
+
+    # add handlers to logger
+    logger = logging.getLogger()
+    logger.addHandler(file_handler)
+    logger.addHandler(stream_handler)
+
+    # 0 = no debug message
+
+
+    if level == "1":
+        #logger.setLevel(logging.ERROR)
+        stream_handler.setLevel(logging.ERROR)
+        file_handler.setLevel(logging.ERROR)
+
+    if level == "2":
+        #logger.setLevel(logging.WARNING)
+        stream_handler.setLevel(logging.ERROR)
+        file_handler.setLevel(logging.WARNING)
+
+    if level == "3":
+        #logger.setLevel(logging.CRITICAL)
+        stream_handler.setLevel(logging.WARNING)
+        file_handler.setLevel(logging.DEBUG)
+
+    if level == "0" or level == None:
+        #logger.setLevel(logging.CRITICAL)
+        stream_handler.setLevel(logging.CRITICAL)
+        file_handler.setLevel(logging.CRITICAL)
+
+  #  logger.setLevel(logging.DEBUG)
+    logger.addHandler(file_handler)
+    logger.addHandler(stream_handler)
+
+
+def parse_cmd_arguments():
+    parser = argparse.ArgumentParser(description='Process some integers.')
+    parser.add_argument('-i', '--input', help='input JSON file', required=True)
+    parser.add_argument('-o', '--output', help='output JSON file', required=True)
+    parser.add_argument('-d', '--debug', help='set logging level', required=False)
+
+    return parser.parse_args()
+
+
+def load_rentals_file(filename):
+    with open(filename) as file:
+        try:
+            data = json.load(file)
+        except Exception as exception:
+            logging.critical(f'Unable to load json file with name {filename} due to {exception}')
+            exit(0)
+        return data
+
+    # with open(filename, 'r') as file:
+    #     loaded_json_data = json.load(file)
+    #     try:
+    #         data = json.load(file)
+    #         return data
+    #     except Exception as exception:
+    #         logging.error(exception)
+    #         logging.debug('type: {}, {}'.format(type(d), len(d)))
+    # exit(-1)
+    # return loaded_json_data
+
+def calculate_additional_fields(data):
+    output = []
+
+    for key, value in data.items():
+        try:
+            if value["rental_start"] == "" or value["rental_end"] == "":
+                #logging
+                continue
+            rental_start = datetime.datetime.strptime(value["rental_start"], '%m/%d/%y')
+            rental_end = datetime.datetime.strptime(value["rental_end"], '%m/%d/%y')
+            value["total_days"] = abs((rental_end - rental_start).days)
+            value["total_price"] = value["total_days"] * value["price_per_day"]
+            value["sqrt_total_price"] = math.sqrt(value["total_price"])
+            value["unit_cost"] = value["total_price"] / value["units_rented"]
+
+            value["key"] = key
+
+            output.append(value)
+
+        except Exception as e:
+            print(e)
+
+            #     # ValueError as ex:
```

Comment:
In general, you should try to delete all commented code since they are not being used